python/vespa/vespa/package.py

```diff
@@ -468,3 +490,160 @@ def deploy(self, disk_folder: str, container_memory: str = "4G"):
+        cert_file.write(cert.public_bytes(serialization.Encoding.PEM).decode('UTF-8'))
+        cert_file.flush()
+        return cert_file
+
+    @staticmethod
+    def create_certificate_pair() -> (ec.EllipticCurvePrivateKey, x509.Certificate):
+        key = ec.generate_private_key(ec.SECP521R1, default_backend())
+        name = x509.Name([x509.NameAttribute(x509.NameOID.COMMON_NAME, u'localhost')])
+        certificate = x509.CertificateBuilder() \
+            .subject_name(name) \
+            .issuer_name(name) \
+            .serial_number(x509.random_serial_number()) \
+            .not_valid_before(datetime.utcnow() - timedelta(minutes = 1)) \
+            .not_valid_after(datetime.utcnow() + timedelta(days = 7)) \
+            .public_key(key.public_key()) \
+            .sign(key, hashes.SHA256(), default_backend())
+        return (key, certificate)
+
+    def request(self, method: str, path: str, body: BytesIO = BytesIO(), headers = {}) -> dict:
+        digest = hashes.Hash(hashes.SHA256(), default_backend())
+        body.seek(0)
+        digest.update(body.read())
+        content_hash = standard_b64encode(digest.finalize()).decode('UTF-8')
+        timestamp = datetime.utcnow().isoformat() + 'Z' # Java's Instant.parse requires the neutral time zone appended
+        url = 'https://' + self.connection.host + ":" + str(self.connection.port) + path
+
+        canonical_message = method + '\n' + url + '\n' + timestamp + '\n' + content_hash
+        signature = self.api_key.sign(canonical_message.encode('UTF-8'), ec.ECDSA(hashes.SHA256()))
+
+        headers = {
+            "X-Timestamp": timestamp,
+            "X-Content-Hash": content_hash,
+            "X-Key-Id": self.tenant + ':' + self.application + ':' + "default",
+            "X-Key": self.api_public_key_bytes,
+            "X-Authorization": standard_b64encode(signature),
+            **headers
+        }
+
+        body.seek(0)
+        self.connection.request(method, path, body, headers)
+        with self.connection.getresponse() as response:
+            parsed = json.load(response)
+            if response.status != 200:
+                raise RuntimeError("Status code " + str(response.status) + " doing " + method + " at " + url + ":\n" + parsed['message'])
+            return parsed
+
+    def get_dev_region(self) -> str:
+        return self.request('GET', '/zone/v1/environment/dev/default')['name']
+
+    def get_endpoint(self, instance: str, region: str) -> str:
+        endpoints = self.request('GET',
+                                 '/application/v4/tenant/{}/application/{}/instance/{}/environment/dev/region/{}' \
+                                 .format(self.tenant, self.application, instance, region))['endpoints']
+        container_url = [endpoint['url'] for endpoint in endpoints if endpoint['cluster'] == 'test_app_container']
```

Comment:
Hard-coded container name here. Remember this is the container name changes (in `services.xml`).